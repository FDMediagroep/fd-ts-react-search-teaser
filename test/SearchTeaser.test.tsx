import TestRenderer from 'react-test-renderer';
import SearchTeaser from '../src/SearchTeaser';
import React from 'react';

describe('Search Teaser', () => {
    const teaserData: any = {
        id: "1286721",
        baseUrl: "https://fd.nl",
        url: "https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe",
        image: {
            src: 'https://images.fd.nl/h7f0-gZo7t8HbiVhcWgo5x8HPaA.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect=',
            alt: 'Amsterdam doet ertoe'
        },
        sourceSets: [{
            media: '(max-width: 640px)',
            srcSet: `https://images.fd.nl/h7f0-gZo7t8HbiVhcWgo5x8HPaA.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect=`
        }, {
            media: '(max-width: 860px)',
            srcSet: `https://images.fd.nl/h7f0-gZo7t8HbiVhcWgo5x8HPaA.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=115&h=115&rect=`
        }, {
            media: '(min-width: 861px)',
            srcSet: `https://images.fd.nl/h7f0-gZo7t8HbiVhcWgo5x8HPaA.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=115&h=115&rect=`
        }],
        figCaption: "Amsterdam doet ertoe",
        subject: "Buitenland",
        comments: 1,
        readableAge: "34 min",
        date: "17 Nov 1981",
        title: "Amsterdam doet ertoe",
        printTitle: "Amsterdam doet ertoe in de krant",
        description: "Een wethouder Verkeer en Vervoer, in Davos? Zeker! Wethouder Sharon Dijksma legt op het World Economic Forum uit hoe Amsterdam omgaat met duurzame mobiliteit."
    };

    test('renders correctly', () => {
        let teaser = TestRenderer.create(<SearchTeaser {...teaserData}/>);
        expect(teaser.toJSON()).toMatchSnapshot();
    });

    test('renders bookmarked correctly', () => {
        let teaser = TestRenderer.create(<SearchTeaser {...teaserData} bookmarked={true}/>);
        expect(teaser.toJSON()).toMatchSnapshot();
    });

    test('renders multiple comments correctly', () => {
        let teaser = TestRenderer.create(<SearchTeaser {...teaserData} comments={99}/>);
        expect(teaser.toJSON()).toMatchSnapshot();
    });
});
