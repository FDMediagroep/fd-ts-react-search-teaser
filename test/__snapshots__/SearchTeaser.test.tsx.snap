// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Teaser renders bookmarked correctly 1`] = `
<div
  cardStyle="default"
  className="fd-card fd-search-teaser default"
  id="1286721"
>
  <a
    href="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
  >
    <figure
      className="fd-teaser-figure"
    >
      <picture>
        <source
          media="(max-width: 640px)"
          srcSet="https://images.fd.nl/h7f0-gZo7t8HbiVhcWgo5x8HPaA.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect="
        />
        <source
          media="(max-width: 860px)"
          srcSet="https://images.fd.nl/h7f0-gZo7t8HbiVhcWgo5x8HPaA.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=115&h=115&rect="
        />
        <source
          media="(min-width: 861px)"
          srcSet="https://images.fd.nl/h7f0-gZo7t8HbiVhcWgo5x8HPaA.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=115&h=115&rect="
        />
        <img
          alt="Amsterdam doet ertoe"
          src="https://images.fd.nl/h7f0-gZo7t8HbiVhcWgo5x8HPaA.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect="
        />
      </picture>
    </figure>
    <div
      className="meta"
    >
      <span
        className="prefix"
      >
        Buitenland
      </span>
      <time>
        34 min
      </time>
    </div>
    <div
      className="text-container"
    >
      <h1
        className="fd-typography search-teaser-h"
        textStyle="search-teaser-h"
      >
        Amsterdam doet ertoe
      </h1>
      <p
        className="print-title"
      >
        <span>
          Krantentitel: 
        </span>
        <span
          className="print-title-text"
        >
          ‘
          Amsterdam doet ertoe in de krant
          ’
        </span>
      </p>
      <p
        className="intro"
      >
        Een wethouder Verkeer en Vervoer, in Davos? Zeker! Wethouder Sharon Dijksma legt op het World Economic Forum uit hoe Amsterdam omgaat met duurzame mobiliteit.
      </p>
    </div>
  </a>
  <footer
    className="fd-teaser-footer"
  >
    <menu>
      <span
        className="placeholder"
      />
      <a
        className="share-social"
        href="javascript: void(0);"
      >
        <i
          className="icon-share-alt"
          data-id="1286721"
          data-section="Buitenland"
          data-title="Amsterdam doet ertoe"
          data-url="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
          title="Delen"
        />
        <span
          className="icon-label"
        >
          Delen
        </span>
      </a>
      <a
        className="favorite selected"
        data-addurl="/add-favorite"
        data-articleid="1286721"
        data-deleteurl="/delete-favorite"
        data-key="1286721"
        data-title="Amsterdam doet ertoe"
        data-url="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
        href="javascript: void(0);"
        id="favorite1286721"
        title="Bewaren"
      >
        <i
          className="icon-bookmark"
          title="Bewaren"
        />
        <i
          className="icon-bookmark1"
          title="Verwijder van bewaarlijst"
        />
        <span
          className="icon-label"
        >
          Bewaren
        </span>
      </a>
      <a
        className="comments-link"
        data-articleid="1286721"
        href="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe#comments"
      >
        <i
          className="icon-comment-alt"
        />
        <span
          className="comments-count"
        >
          1
        </span>
        <span
          className="comments-suffix"
        >
          reactie
        </span>
      </a>
    </menu>
  </footer>
</div>
`;

exports[`Search Teaser renders correctly 1`] = `
<div
  cardStyle="default"
  className="fd-card fd-search-teaser default"
  id="1286721"
>
  <a
    href="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
  >
    <figure
      className="fd-teaser-figure"
    >
      <picture>
        <source
          media="(max-width: 640px)"
          srcSet="https://images.fd.nl/h7f0-gZo7t8HbiVhcWgo5x8HPaA.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect="
        />
        <source
          media="(max-width: 860px)"
          srcSet="https://images.fd.nl/h7f0-gZo7t8HbiVhcWgo5x8HPaA.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=115&h=115&rect="
        />
        <source
          media="(min-width: 861px)"
          srcSet="https://images.fd.nl/h7f0-gZo7t8HbiVhcWgo5x8HPaA.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=115&h=115&rect="
        />
        <img
          alt="Amsterdam doet ertoe"
          src="https://images.fd.nl/h7f0-gZo7t8HbiVhcWgo5x8HPaA.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect="
        />
      </picture>
    </figure>
    <div
      className="meta"
    >
      <span
        className="prefix"
      >
        Buitenland
      </span>
      <time>
        34 min
      </time>
    </div>
    <div
      className="text-container"
    >
      <h1
        className="fd-typography search-teaser-h"
        textStyle="search-teaser-h"
      >
        Amsterdam doet ertoe
      </h1>
      <p
        className="print-title"
      >
        <span>
          Krantentitel: 
        </span>
        <span
          className="print-title-text"
        >
          ‘
          Amsterdam doet ertoe in de krant
          ’
        </span>
      </p>
      <p
        className="intro"
      >
        Een wethouder Verkeer en Vervoer, in Davos? Zeker! Wethouder Sharon Dijksma legt op het World Economic Forum uit hoe Amsterdam omgaat met duurzame mobiliteit.
      </p>
    </div>
  </a>
  <footer
    className="fd-teaser-footer"
  >
    <menu>
      <span
        className="placeholder"
      />
      <a
        className="share-social"
        href="javascript: void(0);"
      >
        <i
          className="icon-share-alt"
          data-id="1286721"
          data-section="Buitenland"
          data-title="Amsterdam doet ertoe"
          data-url="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
          title="Delen"
        />
        <span
          className="icon-label"
        >
          Delen
        </span>
      </a>
      <a
        className="favorite"
        data-addurl="/add-favorite"
        data-articleid="1286721"
        data-deleteurl="/delete-favorite"
        data-key="1286721"
        data-title="Amsterdam doet ertoe"
        data-url="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
        href="javascript: void(0);"
        id="favorite1286721"
        title="Bewaren"
      >
        <i
          className="icon-bookmark"
          title="Bewaren"
        />
        <i
          className="icon-bookmark1"
          title="Verwijder van bewaarlijst"
        />
        <span
          className="icon-label"
        >
          Bewaren
        </span>
      </a>
      <a
        className="comments-link"
        data-articleid="1286721"
        href="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe#comments"
      >
        <i
          className="icon-comment-alt"
        />
        <span
          className="comments-count"
        >
          1
        </span>
        <span
          className="comments-suffix"
        >
          reactie
        </span>
      </a>
    </menu>
  </footer>
</div>
`;

exports[`Search Teaser renders multiple comments correctly 1`] = `
<div
  cardStyle="default"
  className="fd-card fd-search-teaser default"
  id="1286721"
>
  <a
    href="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
  >
    <figure
      className="fd-teaser-figure"
    >
      <picture>
        <source
          media="(max-width: 640px)"
          srcSet="https://images.fd.nl/h7f0-gZo7t8HbiVhcWgo5x8HPaA.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect="
        />
        <source
          media="(max-width: 860px)"
          srcSet="https://images.fd.nl/h7f0-gZo7t8HbiVhcWgo5x8HPaA.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=115&h=115&rect="
        />
        <source
          media="(min-width: 861px)"
          srcSet="https://images.fd.nl/h7f0-gZo7t8HbiVhcWgo5x8HPaA.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=115&h=115&rect="
        />
        <img
          alt="Amsterdam doet ertoe"
          src="https://images.fd.nl/h7f0-gZo7t8HbiVhcWgo5x8HPaA.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect="
        />
      </picture>
    </figure>
    <div
      className="meta"
    >
      <span
        className="prefix"
      >
        Buitenland
      </span>
      <time>
        34 min
      </time>
    </div>
    <div
      className="text-container"
    >
      <h1
        className="fd-typography search-teaser-h"
        textStyle="search-teaser-h"
      >
        Amsterdam doet ertoe
      </h1>
      <p
        className="print-title"
      >
        <span>
          Krantentitel: 
        </span>
        <span
          className="print-title-text"
        >
          ‘
          Amsterdam doet ertoe in de krant
          ’
        </span>
      </p>
      <p
        className="intro"
      >
        Een wethouder Verkeer en Vervoer, in Davos? Zeker! Wethouder Sharon Dijksma legt op het World Economic Forum uit hoe Amsterdam omgaat met duurzame mobiliteit.
      </p>
    </div>
  </a>
  <footer
    className="fd-teaser-footer"
  >
    <menu>
      <span
        className="placeholder"
      />
      <a
        className="share-social"
        href="javascript: void(0);"
      >
        <i
          className="icon-share-alt"
          data-id="1286721"
          data-section="Buitenland"
          data-title="Amsterdam doet ertoe"
          data-url="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
          title="Delen"
        />
        <span
          className="icon-label"
        >
          Delen
        </span>
      </a>
      <a
        className="favorite"
        data-addurl="/add-favorite"
        data-articleid="1286721"
        data-deleteurl="/delete-favorite"
        data-key="1286721"
        data-title="Amsterdam doet ertoe"
        data-url="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
        href="javascript: void(0);"
        id="favorite1286721"
        title="Bewaren"
      >
        <i
          className="icon-bookmark"
          title="Bewaren"
        />
        <i
          className="icon-bookmark1"
          title="Verwijder van bewaarlijst"
        />
        <span
          className="icon-label"
        >
          Bewaren
        </span>
      </a>
      <a
        className="comments-link"
        data-articleid="1286721"
        href="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe#comments"
      >
        <i
          className="icon-comment-alt"
        />
        <span
          className="comments-count"
        >
          99
        </span>
        <span
          className="comments-suffix"
        >
          reacties
        </span>
      </a>
    </menu>
  </footer>
</div>
`;
